grpc/server/dronecore_server.cpp

```diff
@@ -111,25 +140,59 @@ class DroneCoreRPCImpl final : public DroneCoreRPC::Service
         return Status::OK;
     }
 
-    DroneCore dc;
-
 private:
 };
 
+class TelemetryRPCImpl final : public TelemetryRPC::Service
+{
+
+    Status TelemetryPositionSubscription(ServerContext *context, const TelemetryEmpty *request,
+                                         ServerWriter<dronecorerpc::TelemetryPosition> *writer) override
+    {
+        dc.device().telemetry().position_async([&writer](Telemetry::Position position) {
+            dronecorerpc::TelemetryPosition rpc_position;
+            rpc_position.set_latitude_deg(position.latitude_deg);
+            rpc_position.set_longitude_deg(position.longitude_deg);
+            rpc_position.set_relative_altitude_m(position.relative_altitude_m);
+            rpc_position.set_absolute_altitude_m(position.absolute_altitude_m);
+            writer->Write(rpc_position);
+        });
+        // TODO: This is probably not the best idea.
+        while (true) {
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+        }
+        return Status::OK;
+    }
+};
+
 void RunServer()
 {
     std::string server_address("0.0.0.0:50051");
     DroneCoreRPCImpl service;
 
+    map_type map;
+    map["action"] = &createInstances<ActionRPCImpl>;
+    map["mission"] = &createInstances<MissionRPCImpl>;
+    map["telemetry"] = &createInstances<TelemetryRPCImpl>;
```

Comment:
This will need to be auto-generated in the same way as the plugins are currently added to device_impl.

Reply:
Do you agree?

Reply:
@julianoes createInstance takes class name as parameter, if not here somewhere there should be mapping between string to class name. Any other way we could do without hardcoded mapping.

Thanks for the review, we will address other comments in next submission.